src/app-layout/utils/use-pointer-events.ts

```diff
@@ -1,49 +1,62 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 import { useCallback } from 'react';
-import { SizeControlProps } from '../interfaces';
 import styles from '../styles.css.js';
 
+export interface SizeControlProps {
+  position: 'side' | 'bottom';
+  panelRef?: React.RefObject<HTMLDivElement>;
+  handleRef?: React.RefObject<HTMLDivElement>;
+  setSidePanelWidth: (width: number) => void;
+  setBottomPanelHeight: (height: number) => void;
+}
+
 export const usePointerEvents = ({
   position,
-  splitPanelRef,
+  panelRef,
   handleRef,
   setSidePanelWidth,
   setBottomPanelHeight,
 }: SizeControlProps) => {
   const onDocumentPointerMove = useCallback(
     (event: PointerEvent) => {
-      if (!splitPanelRef || !splitPanelRef.current || !handleRef || !handleRef.current) {
+      if (!panelRef || !panelRef.current || !handleRef || !handleRef.current) {
         return;
       }
 
+      panelRef.current.style.transitionProperty = 'none';
+
       if (position === 'side') {
         const mouseClientX = event.clientX;
 
         // The handle offset aligns the cursor with the middle of the resize handle.
         const handleOffset = handleRef.current.getBoundingClientRect().width / 2;
-        const width = splitPanelRef.current.getBoundingClientRect().right - mouseClientX + handleOffset;
+        const width = panelRef.current.getBoundingClientRect().right - mouseClientX + handleOffset;
 
         setSidePanelWidth(width);
       } else {
         const mouseClientY = event.clientY;
 
         // The handle offset aligns the cursor with the middle of the resize handle.
         const handleOffset = handleRef.current.getBoundingClientRect().height / 2;
-        const height = splitPanelRef.current.getBoundingClientRect().bottom - mouseClientY + handleOffset;
+        const height = panelRef.current.getBoundingClientRect().bottom - mouseClientY + handleOffset;
 
         setBottomPanelHeight(height);
       }
     },
-    [position, splitPanelRef, handleRef, setSidePanelWidth, setBottomPanelHeight]
+    [position, panelRef, handleRef, setSidePanelWidth, setBottomPanelHeight]
   );
 
   const onDocumentPointerUp = useCallback(() => {
+    if (!panelRef || !panelRef.current) {
+      return;
+    }
     document.body.classList.remove(styles['resize-active']);
     document.body.classList.remove(styles[`resize-${position}`]);
     document.removeEventListener('pointerup', onDocumentPointerUp);
     document.removeEventListener('pointermove', onDocumentPointerMove);
-  }, [onDocumentPointerMove, position]);
+    panelRef.current.style.transitionProperty = 'border-color, opacity, width';
```

Comment:
Do these three properties correspond to something in a CSS file? I see a risk that the definition here might get out of sync with the CSS, but also at runtime it might get out of sync with React's virtual DOM, since imperative updating of DOM elements happens without React's knowledge. We could make this more reliable and maintainable by setting and removing a class on the DOM element instead of inline styles (e.g. something like `panelRef.current.classList.add(styles.disableTransitions)`)

Reply:
This is such a good idea! I hadn't thought of that.